base/set.jl

```diff
@@ -476,10 +478,20 @@ allunique(::Union{AbstractSet,AbstractDict}) = true
 
 allunique(r::AbstractRange) = !iszero(step(r)) || length(r) <= 1
 
-allunique(A::StridedArray) = length(A) < 32 ? _indexed_allunique(A) : _hashed_allunique(A)
+function allunique(S::String)
+    nunits = ncodeunits(S) # computing `length` for a string might be expensive
+    nunits < 32 ? _indexed_allunique(S, nunits) : _hashed_allunique(S, nunits)
```

Comment:
At first glance this looks dangerous, but `_indexed_allunique` doesn't actually index. I wonder if it should be called `_iterate_allunique` or `_quadratic_allunique` or something?